stubs/docker/docker/client.pyi

```diff
@@ -12,11 +14,25 @@ from docker.models.services import ServiceCollection
 from docker.models.swarm import Swarm
 from docker.models.volumes import VolumeCollection
 
+@type_check_only
+class _Environ(Protocol):
+    def __getitem__(self, k: str, /) -> str: ...
+    def keys(self) -> Iterable[str]: ...
+
+@type_check_only
+class _FromEnvDict(TypedDict):
+    version: NotRequired[str]
+    timeout: NotRequired[int]
+    max_pool_size: NotRequired[int]
+    environment: NotRequired[_Environ]
+    credstore_env: NotRequired[dict[str, str]]
+    use_ssh_client: NotRequired[bool]
+
 class DockerClient:
     api: APIClient
     def __init__(self, *args, **kwargs) -> None: ...
     @classmethod
-    def from_env(cls, **kwargs) -> DockerClient: ...
+    def from_env(cls, **kwargs: Unpack[_FromEnvDict]) -> DockerClient: ...
```

Comment:
It's probably easier to just list the allowed arguments here, instead of using `Unpack`. This is how we usually handle cases like this in typeshed.

Reply:
Thank you for the speedy response (as often!).

Would that mean, for example: `def from_env(cls, version: str | None = None, ...)`?

Reply:
Exactly.

Reply:
Thank you. I made this change - however, unlike my sample above, I used a `*` to mark the arguments as keyword-only. This was to avoid errors like `error: docker.client.DockerClient.from_env is inconsistent, runtime does not have argument "use_ssh_client". Maybe you forgot to make it keyword-only in the stub?` in CI.

Reply:
Good point, using the star is correct, of course.